services/compute/virtualmachine/virtualmachine.go

```diff
@@ -144,14 +144,36 @@ func (c *client) getWssdVirtualMachineStorageConfigurationImageReference(s *comp
 	return *s.Name, nil
 }
 func (c *client) getWssdVirtualMachineStorageConfigurationOsDisk(s *compute.OSDisk) (*wssdcloudcompute.Disk, error) {
-	if s.Vhd == nil {
-		return nil, errors.Wrapf(errors.InvalidInput, "Vhd Configuration is missing in OSDisk")
+	if s.ManagedDisk == nil {
+		if s.Vhd == nil {
+			return nil, errors.Wrapf(errors.InvalidInput, "Vhd Configuration is missing in OSDisk")
+		}
+		if s.Vhd.URI == nil {
+			return nil, errors.Wrapf(errors.InvalidInput, "Vhd URI Configuration is missing in OSDisk")
+		}
 	}
-	if s.Vhd.URI == nil {
-		return nil, errors.Wrapf(errors.InvalidInput, "Vhd URI Configuration is missing in OSDisk")
+
+	diskName := ""
+	if s.Vhd != nil && s.Vhd.URI != nil {
+		diskName = *s.Vhd.URI
+	}
+	var managedDisk *wssdcommon.VirtualMachineManagedDiskParameters
+	if s.ManagedDisk != nil {
+		managedDisk = &wssdcommon.VirtualMachineManagedDiskParameters{}
+		if s.ManagedDisk.SecurityProfile != nil {
+			securityEncryptionType := wssdcommon.SecurityEncryptionTypes_SecurityEncryptionNone
```

Comment:
>securityEncryptionType := wssdcommon.SecurityEncryptionTypes_SecurityEncryptionNone [](http://example.com/codeflow?start=3&length=83)

should we do this outside the if statement, so that its always set?

Reply:
we could, but we only construct the VMDisksecurityProfile inside the if, so we wouldn't use the securityEncryptionType outside the if block

Reply:
nvm not sure that make sense.